src/journal-remote/journal-upload-journal.c

```diff
@@ -303,6 +301,16 @@ static size_t journal_input_callback(void *buf, size_t size, size_t nmemb, void
         return filled;
 }
 
+static size_t journal_input_callback(void *buf, size_t size, size_t nmemb, void *userp) {
+        assert(nmemb < SSIZE_MAX / size);
+        assert(!size_multiply_overflow(size, nmemb));
+        int r = journal_input_data(buf, size * nmemb, 0, userp);
```

Comment:
"int r" always gets special treatment in our codebase, because we universally use such a local variable to store function error return codes in. Hence, we always declare it separately at the top of the function.

or in other words, do `int r;` at the top of the function